Source/Scene/Scene.js

```diff
@@ -3886,12 +4064,92 @@ define([
         }
         //>>includeEnd('debug');
         var ray = getRayForClampToHeight(this, cartesian);
-        var pickResult = pickFromRay(this, ray, objectsToExclude, true);
+        var pickResult = pickFromRay(this, ray, objectsToExclude, true, false);
         if (defined(pickResult)) {
             return Cartesian3.clone(pickResult.position, result);
         }
     };
 
+    /**
+     * Initiates an asynchronous {@link Scene#sampleHeight} query for an array of {@link Cartographic} positions
```

Comment:
Should this and any other similar functions explicitly mention that the input `height` of each position is ignored?

Reply:
Are you sure the height should be set to `undefined` if it can't be found?  Seems like robust code will now always have to check for `undefined`.  I almost wonder if it should set to `0.0` and return a separate array or something for positions that couldn't be resolved.  I dunno.  Think through how simple and robust apps will use the returned values.

Reply:
I sort of agree. I tried to keep this function consistent with `sampleTerrain`, which also sets the height to undefined.

Another approach I was considering initially was to only take a single cartographic rather than an array. The caller would be responsible for calling it multiple times and managing the promises.

Reply:
Gotcha - could also do nothing for now if no one has complained about `sampleTerrain`, which I think is pretty widely used.